api/src/main/java/com/codeforcommunity/rest/subrouter/TeamsRouter.java

```diff
@@ -22,12 +26,71 @@ public TeamsRouter(ITeamsProcessor processor) {
   @Override
   public Router initializeRouter(Vertx vertx) {
     Router router = Router.router(vertx);
-
+    registerCreateTeam(router);
+    registerGetTeam(router);
+    registerAddGoal(router);
+    registerDeleteGoal(router);
     registerDisbandTeam(router);
 
     return router;
   }
 
+  // Create a Team
+  private void registerCreateTeam(Router router) {
+    Route createTeamRoute = router.post("/create");
+    createTeamRoute.handler(this::handleCreateTeamRoute);
+  }
+
+  private void handleCreateTeamRoute(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    CreateTeamRequest createTeamRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, CreateTeamRequest.class);
+    processor.createTeam(userData, createTeamRequest);
+    end(routingContext.response(), 200);
+  }
+
+  // Get a Team
+  private void registerGetTeam(Router router) {
+    Route getTeamRoute = router.get("/:team_id");
+    getTeamRoute.handler(this::handleGetTeamRoute);
+  }
+
+  private void handleGetTeamRoute(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jwt_data");
+    GetTeamRequest getTeamRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, GetTeamRequest.class);
+    processor.getTeam(userData, getTeamRequest);
+    end(routingContext.response(), 200);
+  }
+
+  // Add a Goal
+  private void registerAddGoal(Router router) {
+    Route addGoalRoute = router.post("/:team_id/add_goal");
+    addGoalRoute.handler(this::handleAddGoal);
+  }
+
+  private void handleAddGoal(RoutingContext routingContext) {
+    JWTData userData = routingContext.get("jtw_data");
+    AddGoalRequest addGoalRequest =
+        RestFunctions.getJsonBodyAsClass(routingContext, AddGoalRequest.class);
+    processor.addGoal(userData, addGoalRequest);
```

Comment:
Once you take the teamId out of the DTO then you'll have to have another line getting the team_id from the path (it's the same as in the getTeam route) and then passing it to the processor along with the DTO.